/* 
 好友操作服务器的子服务注册信息： /service/friend/instance_id 
 服务名称：/service/friend 
 实例 ID: instance_id 每个能够提供用户操作服务的子服务器唯
一 ID 
 当服务发现的时候，通过 /service/friend 进行服务发现，就可以发现所
有的能够提供用户操作的实例信息了 
*/ 
syntax = "proto3"; 
package chatnow; 
import "base.proto"; 
 option cc_generic_services = true; 
 
//-------------------------------------- 
//好友列表获取 
message GetFriendListReq { 
    string request_id = 1; 
    optional string user_id = 2; 
    optional string session_id = 3; 
} 
message GetFriendListRsp { 
    string request_id = 1; 
    bool success = 2; 
    string errmsg = 3; 
    repeated UserInfo friend_list = 4; 
} 
 
//-------------------------------------- 
//好友删除 
message FriendRemoveReq { 
    string request_id = 1; 
    optional string user_id = 2; 
    optional string session_id = 3; 
    string peer_id = 4; 
} 
message FriendRemoveRsp { 
    string request_id = 1; 
    bool success = 2; 
    string errmsg = 3; 
} 
//-------------------------------------- 
//添加好友--发送好友申请 
message FriendAddReq { 
    string request_id = 1; 
    optional string session_id = 2; 
    optional string user_id = 3;//申请人 id 
    string respondent_id = 4;//被申请人 id 
} 
message FriendAddRsp { 
    string request_id = 1; 
    bool success = 2; 
    string errmsg = 3; 
    optional string notify_event_id = 4;//通知事件 id 
} 
//-------------------------------------- //好友申请的处理 
message FriendAddProcessReq { 
    string request_id = 1; 
    string notify_event_id = 2;//通知事件 id 
    bool agree = 3;//是否同意好友申请 
    string apply_user_id = 4; //申请人的用户 id 
    optional string session_id = 5; 
    optional string user_id = 6; 
} 
// +++++++++++++++++++++++++++++++++ 
message FriendAddProcessRsp { 
    string request_id = 1; 
    bool success = 2; 
    string errmsg = 3; 
    // 同意后会创建会话，向网关返回会话信息，用于通知双方会话的建立，这个字段客户端不需要关注 
    optional string new_session_id = 4; 
} 
//-------------------------------------- 
//获取待处理的，申请自己好友的信息列表 
message GetPendingFriendEventListReq { 
    string request_id = 1; 
    optional string session_id = 2; 
    optional string user_id = 3; 
} 
 
message FriendEvent { 
    optional string event_id = 1; 
    UserInfo sender = 3; 
} 
message GetPendingFriendEventListRsp { 
    string request_id = 1; 
    bool success = 2; 
    string errmsg = 3; 
    repeated FriendEvent event = 4; 
} 
 
//-------------------------------------- 
//好友搜索 
message FriendSearchReq { 
    string request_id = 1; 
    string search_key = 2;//就是名称模糊匹配关键字 
    optional string session_id = 3;  
    optional string user_id = 4; 
} 
message FriendSearchRsp { 
    string request_id = 1; 
    bool success = 2; 
    string errmsg = 3; 
    repeated UserInfo user_info = 4; 
} 
 
 
service FriendService { 
 rpc GetFriendList(GetFriendListReq) returns (GetFriendListRsp); 
 rpc FriendRemove(FriendRemoveReq) returns (FriendRemoveRsp); 
 rpc FriendAdd(FriendAddReq) returns (FriendAddRsp); 
 rpc FriendAddProcess(FriendAddProcessReq) returns (FriendAddProcessRsp); 
 rpc FriendSearch(FriendSearchReq) returns (FriendSearchRsp); 
 rpc GetPendingFriendEventList(GetPendingFriendEventListReq) returns (GetPendingFriendEventListRsp); 
}